vkcs/mlplatform/resource_spark_k8s.go

```diff
@@ -0,0 +1,483 @@
+			},
+
+			"name": schema.StringAttribute{
+				Required:    true,
+				Description: "Cluster name. Changing this creates a new resource",
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+
+			"availability_zone": schema.StringAttribute{
+				Required:    true,
+				Description: "The availability zone in which to create the resource. Changing this creates a new resource",
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+
+			"network_id": schema.StringAttribute{
+				Required:    true,
+				Description: "ID of the network. Changing this creates a new resource",
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+
+			"subnet_id": schema.StringAttribute{
+				Optional:    true,
+				Description: "ID of the subnet. Changing this creates a new resource",
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+
+			"node_groups": schema.ListNestedAttribute{
+				NestedObject: schema.NestedAttributeObject{
+					Attributes: map[string]schema.Attribute{
+						"node_count": schema.Int64Attribute{
+							Optional:    true,
+							Description: "Count of nodes in node group",
+						},
+
+						"flavor_id": schema.StringAttribute{
+							Required:    true,
+							Description: "ID of the flavor to be used in nodes",
+						},
+
+						"autoscaling_enabled": schema.BoolAttribute{
+							Required:    true,
+							Description: "Enables autoscaling for node group",
+						},
+
+						"min_nodes": schema.Int64Attribute{
+							Optional:    true,
+							Description: "Minimum count of nodes in node group. It is used only when autoscaling is enabled",
+						},
+
+						"max_nodes": schema.Int64Attribute{
+							Optional:    true,
+							Description: "Maximum number of nodes in node group. It is used only when autoscaling is enabled",
+						},
+					},
+				},
+				Required:    true,
+				Description: "Cluster's node groups configuration",
+			},
+
+			"cluster_mode": schema.StringAttribute{
+				Required:    true,
+				Description: "Cluster Mode. Should be `DEV` or `PROD`. Changing this creates a new resource",
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+				Validators: []validator.String{
+					stringvalidator.OneOf("DEV", "PROD"),
+				},
+			},
+
+			"registry_id": schema.StringAttribute{
+				Required:    true,
+				Description: "ID of the K8S registry to use with cluster. Changing this creates a new resource",
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+
+			"ip_pool": schema.StringAttribute{
+				Required:    true,
+				Description: "ID of the ip pool. Changing this creates a new resource",
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+
+			"keypair": schema.StringAttribute{
```

Comment:
btw are this argument name is the same across all our resources?

Reply:
There is some inconsistency - compute resources use "key_pair", all other (db, k8s) - "keypair"